src/schnetpack/interfaces/ase_interface.py

```diff
@@ -341,6 +341,289 @@ def calculate(
+        }
+
+        # Load multiple models
+        self.models = [self._load_model(model_file) for model_file in models]
+
+        if uncertainty_fn is None:
+            self.uncertainty_fn = self.AbsoluteUncertainty
+        else:
+            self.uncertainty_fn = uncertainty_fn.__get__(self, type(self))
+
+    def calculate(
+        self,
+        atoms: ase.Atoms = None,
+        properties: List[str] = ["energy"],
+        system_changes: List[str] = all_changes,
+        energy_weight: float = 0,
+        force_weight: float = 1.0,
+        stress_weight: float = 0,
+    ):
+        """
+        Calculate properties by averaging results from multiple models.
+        """
+        Calculator.calculate(self, atoms)
+
+        model_inputs = self.converter(atoms)
+        accumulated_results = {prop: [] for prop in properties}
+
+        for model in self.models:
+            model_results = model(model_inputs)
+
+            for prop in properties:
+                model_prop = self.property_map[prop]
+                if model_prop in model_results:
+                    value = model_results[model_prop].cpu().data.numpy()
+                    if prop == self.energy:
+                        value = value.item()
+                    elif prop == self.stress:
+                        value = value.squeeze()
+                    accumulated_results[prop].append(value * self.property_units[prop])
+                else:
+                    raise AtomsConverterError(
+                        f"'{prop}' is not a property of your models. Please check the model properties!"
+                    )
+
+        # Compute average values
+        self.results = {
+            prop: np.mean(accumulated_results[prop], axis=0) for prop in properties
+        }
+        # Compute uncertainty using assigned uncertainty function
+        self.results["uncertainty"] = self.uncertainty_fn(
+            predictions=accumulated_results,
+            energy_weight=energy_weight,
+            force_weight=force_weight,
+            stress_weight=stress_weight,
+        )
+
+    def get_uncertainty(
+        self,
+        atoms,
+        properties: List[str] = None,
+        energy_weight: float = 0,
+        force_weight: float = 1.0,
+        stress_weight: float = 0,
+    ):
+        """
+        Ensure calculation is up to date and return the uncertainty.
+        """
+        if properties is None:
+            properties = [self.energy, self.forces] + (
+                [self.stress] if self.stress_key is not None else []
+            )
+        if self.calculation_required(atoms, properties):
+            self.calculate(
+                atoms,
+                properties,
+                energy_weight=energy_weight,
+                force_weight=force_weight,
+                stress_weight=stress_weight,
+            )
+        return self.results["uncertainty"]
+
+    def AbsoluteUncertainty(
```

Comment:
Make this a class that stores the weights to `self.xxx_weight`. Then use `__call__(self, atoms)` to make the class callable. This way we do not need to pass these args to the `calculate` functions of the calculator